engine/src/tests/test.cpp

```diff
@@ -0,0 +1,9 @@
+#include <gtest/gtest.h>
+
+// Demonstrate some basic assertions.
+TEST(HelloTest, BasicAssertions) {
+  // Expect two strings not to be equal.
+  EXPECT_STRNE("hello", "world");
+  // Expect equality.
+  EXPECT_EQ(7 * 6, 42);
+}
```

Comment:
For comparison Boost::Test equivalent:

```C++
#define BOOST_TEST_DYN_LINK
#define BOOST_TEST_NO_MAIN

#include <boost/test/unit_test.hpp>

BOOST_AUTO_TEST_SUITE( GroupName );

BOOST_AUTO_TEST( IndividualTestName )
{
    // For whatever reason Boost doesn't do a not-equal directly
    BOOST_CHECK("hello" != "world");
    // but they do do equal directly
    BOOST_CHECK_EQUAL(7*6, 42);
}
```

Reply:
Yeah. Like I said, at the basic level, they're all the same and I have looked at Boost.